app/helpers/fhir_helper.rb

```diff
@@ -341,4 +342,47 @@ def from_statelocal_id_extension(patient)
     statelocal_id = patient&.identifier&.find { |i| i&.system == 'http://saraalert.org/SaraAlert/state-local-id' }
     statelocal_id&.value
   end
+
+  def to_address_by_type_extension(patient, address_type)
+    case address_type
+    when 'USA'
+      FHIR::Address.new(
+        line: [patient.address_line_1, patient.address_line_2].reject(&:blank?),
+        city: patient.address_city,
+        district: patient.address_county,
+        state: patient.address_state,
+        postalCode: patient.address_zip
+      )
+    when 'Foreign'
+      [patient.foreign_address_line_1,
+       patient.foreign_address_line_2,
+       patient.foreign_address_line_3,
+       patient.foreign_address_city,
+       patient.foreign_address_country,
+       patient.foreign_address_zip,
+       patient.foreign_address_state].any? ?
+      FHIR::Address.new(
+        line: [patient.foreign_address_line_1, patient.foreign_address_line_2, patient.foreign_address_line_3].reject(&:blank?),
+        city: patient.foreign_address_city,
+        country: patient.foreign_address_country,
+        state: patient.foreign_address_state,
+        postalCode: patient.foreign_address_zip,
+        extension: [FHIR::Extension.new(url: 'http://saraalert.org/StructureDefinition/address-type', valueString: 'Foreign')]
+      ) : nil
+    end
+  end
+
+  def from_address_by_type_extension(patient, address_type)
+    address = (patient&.address&.select do |a|
```

Comment:
Why not a `find` (https://apidock.com/ruby/Enumerable/find) instead of `select` (particularly because you are just grabbing `first` on line 378). That way you could do without the parens.

Reply:
Oh good point, that's cleaner, fixed in https://github.com/SaraAlert/SaraAlert/pull/686/commits/844495ba44025e3f73ba4e5efacabe79cbf09de6.